File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.maxsz = maxSize
        self.st = []

    def push(self, x: int) -> None:
        if len(self.st) < self.maxsz:
            self.st.append(x)

    def pop(self) -> int:
        if len(self.st):
            x = self.st[-1]
            self.st.pop()
            return x
        return -1

    def increment(self, k: int, val: int) -> None:
        x = k
        if(len(self.st) < k):
            x = len(self.st)
        for i in range(x):
            self.st[i] += val



# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)

File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.cpp
class CustomStack {
public:
    vector<int> v;
    int maxsz;
    CustomStack(int maxSize) {
        maxsz = maxSize;
    }
    
    void push(int x) {
        if(maxsz > v.size()) v.push_back(x);
    }
    
    int pop() {
        if(!v.size()) return -1;
        int x = v.back();
        v.pop_back();
        return x;
    }
    
    void increment(int k, int val) {
        int x = k;
        if(v.size() < k) x = v.size();
        for(int i =0; i < x;i++) v[i] +=val;
    }
};

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack* obj = new CustomStack(maxSize);
 * obj->push(x);
 * int param_2 = obj->pop();
 * obj->increment(k,val);
 */